File: Notes/app/src/main/java/com/serhii/apps/notes/database/impl/tables/NoteTableModel.java
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.apps.notes.database.impl.tables;

import android.provider.BaseColumns;

public class NoteTableModel {

    private NoteTableModel() {}

    /**
     *   Db table description for user notes
     */
    public static final class UserNotesEntry implements BaseColumns {
        public static final String TABLE_NAME = "notesList";
        public static final String COLUMN_NAME_NOTE = "note";
    }

    /**
     *  Db table query constants
     */
    public static final String DATABASE_NAME = "notes.db";
    public static final int VERSION = 1;

    /**
     * Db queries
     */
    public static final String QUERY_CREATE_USER_NOTES_TABLE = "CREATE TABLE "
            + UserNotesEntry.TABLE_NAME + " ("
            + UserNotesEntry._ID + " INTEGER PRIMARY KEY AUTOINCREMENT, "
            + UserNotesEntry.COLUMN_NAME_NOTE + " TEXT NOT NULL );";

    public static final String QUERY_DROP_USER_NOTES_TABLE = "DROP TABLE IF EXISTS " + UserNotesEntry.TABLE_NAME;
}

File: Notes/app/src/main/java/com/serhii/apps/notes/database/impl/tables/NoteTableModel.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.apps.notes.database.impl.tables

import android.provider.BaseColumns

object NoteTableModel {
    /**
     * Db table query constants
     */
    const val DATABASE_NAME = "notes.db"
    const val VERSION = 1

    /**
     * Db queries
     */
    const val QUERY_CREATE_TABLE = ("CREATE TABLE "
            + UserNotesEntry.TABLE_NAME + " ("
            + BaseColumns._ID + " INTEGER PRIMARY KEY AUTOINCREMENT, "
            + UserNotesEntry.COLUMN_NAME_NOTE + " TEXT NOT NULL );")

    const val QUERY_CLEAR_TABLE = "DELETE FROM " + UserNotesEntry.TABLE_NAME

    /**
     * Db table description for user notes
     */
    object UserNotesEntry : BaseColumns {
        const val TABLE_NAME = "notesList"
        const val COLUMN_NAME_NOTE = "note"
        const val _ID = "_id"
    }
}